src/io/syncfile.c

```diff
@@ -70,26 +70,22 @@ static MVMint64 mvm_fileno(MVMThreadContext *tc, MVMOSHandle *h) {
  * buffering output. */
 static void perform_write(MVMThreadContext *tc, MVMIOFileData *data, char *buf, MVMint64 bytes) {
     MVMint64 bytes_written = 0;
-    MVM_gc_mark_thread_blocked(tc);
     while (bytes > 0) {
         int r;
-
         do {
+            MVM_gc_mark_thread_blocked(tc);
```

Comment:
Move `MVM_gc_mark_thread_blocked(tc)` into `while (bytes > 0) { ` block may cause a lot of needless marking blocked and marking unblocked